src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.Cache.cs

```diff
@@ -36,14 +36,14 @@ internal sealed partial class JsonClassInfo
 
         // All of the serializable parameters on a POCO constructor keyed on parameter name.
         // Only paramaters which bind to properties are cached.
-        public volatile Dictionary<string, JsonParameterInfo>? ParameterCache;
+        public Dictionary<string, JsonParameterInfo>? ParameterCache;
```

Comment:
These don't need volatile since they are not accessed by other threads until initialization is complete.